crates/edr_provider/src/data.rs

```diff
@@ -92,11 +91,41 @@ pub struct EstimateGasResult {
 
 pub struct SendTransactionResult {
     pub transaction_hash: B256,
-    /// Present if the transaction was auto-mined.
-    pub transaction_result: Option<(ExecutionResult, Trace)>,
     pub mining_results: Vec<DebugMineBlockResult<BlockchainError>>,
 }
 
+impl SendTransactionResult {
+    /// Present if the transaction was auto-mined.
+    pub fn transaction_result_and_trace(&self) -> Option<(&ExecutionResult, &Trace)> {
+        self.mining_results.iter().find_map(|result| {
+            result
+                .transaction_index(&self.transaction_hash)
+                .map(|index| {
+                    (
+                        result.transaction_result(index).expect("index is valid"),
+                        result.transaction_trace(index).expect("index is valid"),
+                    )
+                })
```

Comment:
Request for change:

There is no need to access elements by index as you can use the iterators directly, which voids the need for `expect`. You can reuse the logic that was previously there:

```rust
                        let transaction_result = izip!(
                            result.block.transactions().iter(),
                            result.transaction_results.iter(),
                            result.transaction_traces.iter()
                        )
                        .find_map(|(transaction, result, trace)| {
                            if *transaction.hash() == transaction_hash {
                                Some((result.clone(), trace.clone()))
                            } else {
                                None
                            }
                        });
```

Reply:
Thanks for the feedback! Tbh I prefer to just iterate over one thing and then use the index lookup for the other elements, but it has no effect on performance so I changed it back:

[review: use izip instead of index lookup](https://github.com/NomicFoundation/hardhat/pull/5138/commits/1938735f3d6c3c8b825117e73a7904c6bc6e71e2)